runelite-client/src/main/java/net/runelite/client/plugins/attacklevelrange/AttackLevelRangePlugin.java

```diff
@@ -0,0 +1,135 @@
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+
+@PluginDescriptor(
+	name = "Attack Level Range",
+	description = "Displays range of levels which you can attack in the wilderness on non-PVP worlds.",
+	tags = {"wilderness", "pvp"}
+)
+public class AttackLevelRangePlugin extends Plugin
+{
+	private static final Pattern WILDERNESS_LEVEL_PATTERN = Pattern.compile("^Level: (\\d+)$");
+	private static final int SKULL_CONTAINER_ADJUSTED_RELATIVE_Y = 6;
+	private static final int WILDERNESS_LEVEL_TEXT_ADJUSTED_RELATIVE_Y = 36;
+
+	private String originalWildernessLevelText;
+	private int originalWildernessLevelTextPosition;
+	private int originalSkullContainerPosition;
+
+	@Inject
+	private Client client;
+
+	@Override
+	protected void startUp()
+	{
+		Widget wildernessLevelWidget = client.getWidget(WidgetInfo.PVP_WILDERNESS_LEVEL);
+		if (wildernessLevelWidget != null)
+		{
+			appendAttackLevelRangeText(wildernessLevelWidget);
+		}
+	}
+
+	@Override
+	protected void shutDown()
+	{
+		if (WorldType.isPvpWorld(client.getWorldType()))
+		{
+			return;
+		}
+
+		final Widget wildernessLevelWidget = client.getWidget(WidgetInfo.PVP_WILDERNESS_LEVEL);
+		final Widget skullContainer = client.getWidget(WidgetInfo.PVP_SKULL_CONTAINER);
+
+		if (wildernessLevelWidget != null)
+		{
+			wildernessLevelWidget.setText(originalWildernessLevelText);
+			wildernessLevelWidget.setRelativeY(originalWildernessLevelTextPosition);
+		}
+		if (skullContainer != null)
+		{
+			skullContainer.setRelativeY(originalSkullContainerPosition);
+		}
+
+		originalWildernessLevelText = "";
+		originalWildernessLevelTextPosition = -1;
+		originalSkullContainerPosition = -1;
+	}
+
+	@Subscribe
+	public void onWidgetTextChanged(WidgetTextChanged event)
+	{
+		Widget widget = event.getWidget();
+
+		if (widget.getId() != WidgetInfo.PVP_WILDERNESS_LEVEL.getId())
+		{
+			return;
+		}
+
+		appendAttackLevelRangeText(widget);
+	}
+
+	private void appendAttackLevelRangeText(Widget wildernessLevelWidget)
+	{
+		final String wildernessLevelText = wildernessLevelWidget.getText();
+		final Matcher m = WILDERNESS_LEVEL_PATTERN.matcher(wildernessLevelText);
+		if (!m.matches()
+			|| WorldType.isPvpWorld(client.getWorldType()))
+		{
+			return;
+		}
+
+		originalWildernessLevelText = wildernessLevelText;
+
+		final Widget skullContainer = client.getWidget(WidgetInfo.PVP_SKULL_CONTAINER);
+		if (originalWildernessLevelTextPosition == -1)
+		{
+			originalWildernessLevelTextPosition = wildernessLevelWidget.getRelativeY();
+		}
+		if (originalSkullContainerPosition == -1)
+		{
+			originalSkullContainerPosition = skullContainer.getRelativeY();
+		}
+
+		final int wildernessLevel = Integer.parseInt(m.group(1));
+		final int combatLevel = client.getLocalPlayer().getCombatLevel();
+
+		wildernessLevelWidget.setText(wildernessLevelText + "<br>" + (combatLevel - wildernessLevel) + "-" + (combatLevel + wildernessLevel));
```

Comment:
You might want to check if the upper range is greater than 126 here.

Reply:
Good point. Fixed